pkgs/development/libraries/dbus/default.nix

```diff
@@ -1,92 +1,115 @@
-{ stdenv, lib, fetchurl, pkgconfig, expat, systemd
-, libX11 ? null, libICE ? null, libSM ? null, x11Support ? (stdenv.isLinux || stdenv.isDarwin) }:
+{ stdenv
+, lib
+, fetchurl
+, pkgconfig
+, expat
+, systemd
+, libX11 ? null
+, libICE ? null
+, libSM ? null
+, x11Support ? (stdenv.isLinux || stdenv.isDarwin)
+, dbus
+}:
 
 assert x11Support -> libX11 != null
-                  && libICE != null
-                  && libSM != null;
+  && libICE != null
+  && libSM != null
+;
 
 let
+
+dbus = stdenv.mkDerivation rec {
+  pname = "dbus";
   version = "1.12.16";
-  sha256 = "107ckxaff1cv4q6kmfdi2fb1nlsv03312a7kf6lb4biglhpjv8jl";
-
-self = stdenv.mkDerivation {
-    name = "dbus-${version}";
-    inherit version;
-
-    src = fetchurl {
```

Comment:
If we could preserve indentation, that would make reading this a little bit easier.

Reply:
You can disable whitespace diffing: https://github.com/NixOS/nixpkgs/pull/67363/files?w=1

Reply:
I actually separated the actual changes into the first commits so it could be easy to read.

Reply:
It seems to work even with the dbus passed by the `callPackage` so maybe we can get rid of self after all.